packages/iTwinUI-react/src/core/Slider/Track.tsx

```diff
@@ -42,30 +53,27 @@ export type TrackProps = {
  */
 export const Track = (props: TrackProps) => {
   const { trackDisplayMode, sliderMin, sliderMax, values } = props;
-  const [currentValues, setCurrentValues] = React.useState(
-    [...values].sort((a, b) => a - b),
+  const [segments, setSegments] = React.useState(() =>
+    generateSegments(values, sliderMin, sliderMax),
   );
 
   React.useEffect(() => {
-    const newValues = [...values];
-    newValues.sort((a, b) => a - b);
-    setCurrentValues(newValues);
-  }, [values]);
-
-  const segments = React.useMemo(
-    () => generateSegments(currentValues, sliderMin, sliderMax),
-    [currentValues, sliderMin, sliderMax],
-  );
+    setSegments(generateSegments(values, sliderMin, sliderMax));
+  }, [values, sliderMin, sliderMax]);
 
   return (
     <>
       {'none' !== trackDisplayMode &&
         segments.map((segment, index) => {
           const leftPercent =
-            (100.0 * (segment.left - sliderMin)) / (sliderMax - sliderMin);
-          let rightPercent =
-            (100.0 * (segment.right - sliderMin)) / (sliderMax - sliderMin);
-          rightPercent = 100.0 - rightPercent;
+            segment.left >= sliderMin
+              ? (100.0 * (segment.left - sliderMin)) / (sliderMax - sliderMin)
```

Comment:
Should there be a check here to make sure you don't divide by 0?

Reply:
Added code to avoid divide by case.